mlrun/features.py

```diff
@@ -164,7 +284,112 @@ def check(self, value):
+    ):
+        """Validate min/max length value ranges
+
+        example::
+
+            from mlrun.features import MinMaxLenValidator
+
+            # Add length validator to the feature 'ticker', where valid
+            # minimal length is 1 and maximal length is 10
+            quotes_set["ticker"].validator = MinMaxLenValidator(
+                min=1,
+                max=10,
+                severity="info"
+            )
+
+        :param check_type:  check feature type e.g. True, False
+        :param severity:    severity name e.g. info, warning, etc.
+        :param min:         minimal valid length size
+        :param max:         maximal valid length size
+        """
+        super().__init__(check_type, severity)
+        self.min = min
+        self.max = max
+
+    def check(self, value):
+        ok, args = super().check(value)
+        if ok:
+            if self.min is not None:
+                if len(value) < self.min:
+                    return (
+                        False,
+                        {
+                            "message": "Length value is smaller than min",
+                            "min": self.min,
+                            "length value": len(value),
+                        },
+                    )
+            if self.max is not None:
+                if len(value) > self.max:
+                    return (
+                        False,
+                        {
+                            "message": "Length value is greater than max",
+                            "max": self.max,
+                            "length value": len(value),
+                        },
+                    )
+        return ok, args
+
+
+class RegexValidator(Validator):
+    """Validate value based on regular expression"""
+
+    kind = "regex"
+    _dict_fields = Validator._dict_fields + ["regex"]
+
+    def __init__(self, check_type: bool = None, severity: str = None, regex=None):
+        """Validate value based on regular expression
+
+        example::
+
+            from mlrun.features import RegexValidator
+
+            # Add regular expression validator to the feature 'name' and
+            # expression '(\b[A-Za-z]{1}[0-9]{7}\b)' where valid values are
+            # e.g. A1234567, z9874563, etc.
+            quotes_set["name"].validator = RegexValidator(
+                regex=r"(\b[A-Za-z]{1}[0-9]{7}\b)",
+                severity="info"
+            )
+
+        :param check_type:  check feature type e.g. True, False
+        :param severity:    severity name e.g. info, warning, etc.
+        :param regex:       regular expression for validation
+        """
+        super().__init__(check_type, severity)
+        self.regex = regex
+        self.regex_compile = re.compile(self.regex)
+
+    def check(self, value):
+        ok, args = super().check(value)
+        if ok:
+            if self.regex is not None:
+                if not re.fullmatch(self.regex_compile, value):
```

Comment:
Do we need to consider the possibility that `re.fullmatch(self.regex_compile, value)` will raise an exception? i.e. if value is a number rather than a string.